### PR TITLE
Ability to attach photos

### PR DESCRIPTION
**Pull request by AlinaMoskieva**
Added ability to attach photos to pages as administrator and to comments as user or administrator 💫
@vladimirbazhanov

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def update
+      user.update(user_params)
+
+      if user.save
+        redirect_to admin_users_path
+        flash[:notice] = 'User information was successfully update.'
+      else
+        render :edit
+        flash[:notice] = 'Something wrong.'
+      end
```
Comment (ghost) [inline app/controllers/admin/users_controller.rb:17]: can you pls try to replace if ... else ... end block with respond_to user ?

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    def show
+      return unless current_user.administrator?
+      sign_in(:user, User.find(params[:id]))
+      redirect_to root_url
+      flash[:notice] = "Sign in as #{user.email}."
```
Comment (ghost) [inline app/controllers/admin/users_controller.rb]: Signed in

File: app/controllers/profile_controller.rb [hunk 3]
```diff
@@ -1,7 +1,4 @@
 class ProfileController < ApplicationController
   expose_decorated(:user){ current_user }
   expose(:pages) { user.page.includes(:category).order(created_at: :desc) }
```
Comment (ghost) [inline app/controllers/profile_controller.rb]: user.pages may be?

File: app/models/ckeditor/asset.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
 class Ckeditor::Asset < ActiveRecord::Base
   include Ckeditor::Orm::ActiveRecord::AssetBase
-  include Ckeditor::Backend::Paperclip
+
+  delegate :url, :current_path, :content_type, :to => :data
```
Comment (ghost) [inline app/models/ckeditor/asset.rb]: :to => :data

to: :data

File: app/models/ckeditor/attachment_file.rb [hunk 5]
```diff
@@ -1,11 +1,5 @@
 class Ckeditor::AttachmentFile < Ckeditor::Asset
-  has_attached_file :data,
-                    url: "/ckeditor_assets/attachments/:id/:filename",
-                    path: ":rails_root/public/ckeditor_assets/attachments/:id/:filename"
-
-  validates_attachment_presence :data
-  validates_attachment_size :data, :less_than => 100.megabytes
-  do_not_validate_attachment_file_type :data
+  mount_uploader :data, CkeditorAttachmentFileUploader, :mount_on => :data_file_name
```
Comment (ghost) [inline app/models/ckeditor/attachment_file.rb]: mount_on: :data_file_name

it' ruby 2.x hash syntax

File: app/models/ckeditor/picture.rb [hunk 6]
```diff
@@ -1,12 +1,5 @@
 class Ckeditor::Picture < Ckeditor::Asset
-  has_attached_file :data,
-                    url: "/ckeditor_assets/pictures/:id/:style_:basename.:extension",
-                    path: ":rails_root/public/ckeditor_assets/pictures/:id/:style_:basename.:extension",
-                    styles: { :content => '800>', :thumb => '118x100#' }
-
-  validates_attachment_presence :data
-  validates_attachment_size :data, :less_than => 2.megabytes
-  validates_attachment_content_type :data, :content_type => /\Aimage/
```
Comment (ghost) [inline app/models/ckeditor/picture.rb:9]: and here

File: app/uploaders/ckeditor_attachment_file_uploader.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+class CkeditorAttachmentFileUploader < CarrierWave::Uploader::Base
+  include Ckeditor::Backend::CarrierWave
+
+  # Include RMagick or ImageScience support:
+  # include CarrierWave::RMagick
+  # include CarrierWave::MiniMagick
+  # include CarrierWave::ImageScience
+
+  # Choose what kind of storage to use for this uploader:
+  storage :file
```
Comment (ghost) [inline app/uploaders/ckeditor_attachment_file_uploader.rb:13]: keep in mind that :file storage will not work on heroku
You need to setup Amazon S3 and use it when app in production environment

File: app/policies/administrator_policy.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AdministratorPolicy < ApplicationPolicy
```
Comment (ghost) [inline app/policies/administrator_policy.rb]: мне почему-то кажется, что это должно называться  UserPolicy
Файл с политикой доступа к чему? к Users

File: app/views/pages/index.html.slim [hunk 9]
```diff
@@ -27,5 +27,5 @@
             = " "
             = page.comments.count
       hr/
-  .medium-3.columns data-sticky-container=""
+  .medium-2.columns data-sticky-container=""
     = render "rigth_menu"
```
Comment (ghost) [inline app/views/pages/index.html.slim]: right_menu :)

and rename partial too if it named with error

File: app/views/pages/show.html.slim [hunk 10]
```diff
@@ -10,14 +10,17 @@
         = "Tags:  "
         = link_to(page.get_category.category, pages_path(category_id: page.get_category.id))
     br
-    .medium-8.medium-centered.columns
+    .medium-9.medium-centered.columns
       - policy = PagePolicy.new(current_user, page)
```
Comment (ghost) [inline app/views/pages/show.html.slim:14]: this can be done in controller i think?

**Comment by ghost** [comment]
I like your progress 👍

**Comment by AlinaMoskieva** [comment]
Thank you ☺️

**Pull request closed without merging**